nx/lib/nx/backend.ex

```diff
@@ -204,23 +204,39 @@ defmodule Nx.Backend do
     end
   end
 
-  defp inspect_float(data, 32) do
-    case data do
-      <<0xFF800000::32-native>> -> "-Inf"
-      <<0x7F800000::32-native>> -> "Inf"
-      <<0xFF800001::32-native>> -> "NaN"
-      <<0xFFC00001::32-native>> -> "NaN"
-      <<x::float-32-native>> -> Float.to_string(x)
+  defp inspect_float(<<0xFF800000::32-native>>, 32), do: "-Inf"
+  defp inspect_float(<<0x7F800000::32-native>>, 32), do: "Inf"
+
+  defp inspect_float(<<0x7FF0000000000000::64-native>>, 64), do: "Inf"
+  defp inspect_float(<<0xFFF0000000000000::64-native>>, 64), do: "-Inf"
```

Comment:
So the reason why the code was written with `case` previously is because the pattern matching is more efficient. First we matched on the size and then on the binary. With the current approach, the code may end-up trying the clauses linearly and check the different sizes. Can you please merge them as it was before into a single case? I  know there will be a bit of repetition but I think it is fine. :)